### DISCUSSION TITLE
download previously failed files

### DISCUSSION THREAD

**Discussion by habruening** [abraunegg/onedrive#1689] [Q&A]
I have some files, where the download failed. There was a file download size and hash mis-match. I don't know why this happened. It was an old version of the client.

I have now updated the client to the latest master. Can I tell the client to download the missing files again?

**Comment by abraunegg** [comment]
@habruening 
You need to tell the client to resync via the --resync option.

Additionally, it has been uncovered that some SharePoint sites / OneDrive API is incorrectly providing the file size, thus, this is also a potential for the reason your files did not download.

**Comment by abraunegg** [comment]
@habruening 

Additionally, please can you test the following PR (#1686) to assist with the issue you are facing.

```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/1686/head:pr1686
git checkout pr1686
./configure; make clean; make;
```
To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any options needed> --disable-download-validation
```

When running the PR, your version should be `onedrive v2.4.13-25-ge0f999b`
